test/functional/JIT_Test/src/jit/test/jitserver/JITServerTest.java

```diff
@@ -146,6 +155,15 @@ private static void destroyAndCheckProcess(final Process p, final ProcessBuilder
 	}
 
 	private static Process startProcess(final ProcessBuilder builder, final String name) throws IOException, InterruptedException {
+		// If a previous test did not terminate the process, kill it here (e.g. if assert got triggered before server was terminated)
+		final String commandLine = String.join(" ", builder.command());
+		final String pkillCommandLine[] = {"pkill", "-TERM", "-f", "-x", escapeCommandLineRegexChars(commandLine)};
+		if (Runtime.getRuntime().exec(pkillCommandLine).waitFor() == SUCCESS_RETURN_VALUE) {
+			logger.info("Terminating the previous version of " + name);
+			// if there is a process to kill, wait for it to terminate
+			Thread.sleep(PROCESS_TERMINATE_WAIT_TIME_MS);
+		}
+
```

Comment:
Do any processes get to this point since you added the call to `destroyForcibly()` elsewhere? It seems like there are two possibilities, either a process is gone by now or it's so stuck that another `SIGTERM` wouldn't get rid of it.

Reply:
This would be triggered if a previous test case failed an assert before terminating the server/client

Reply:
Right, makes sense, I recall what you're describing now. This change relies on `name` staying consistent across tests, but I guess it does what's needed and is better in some ways than adding try/catch/kill/rethrow to every test.